src/app/components/core/asset-readings/readings-graph/readings-graph.component.html

```diff
@@ -10,57 +10,65 @@
     </header>
     <section class="modal-card-body">
       <ng-container *ngIf='showGraph'>
-        <div class="columns is-mobile">
-          <div class="column is-two-fifths">
-            <div class="select is-small">
-              <select #selectedTime (change)="getTimeBasedAssetReadingsAndSummary(selectedTime.value)">
-                <!-- <option value="0" [selected]="optedTime == 0">None</option> -->
-                <option value="600" [selected]="optedTime == 600">10 minutes</option>
-                <option value="1800" [selected]="optedTime == 1800">30 minutes</option>
-                <option value="3600" [selected]="optedTime == 3600">1 hour</option>
-                <option value="28800" [selected]="optedTime == 28800">8 hours</option>
-                <option value="86400" [selected]="optedTime == 86400">1 day</option>
-              </select>
+        <div *ngIf="showSpinner" class="app-loading">
+          &nbsp;
+          <svg class="spinner" viewBox="0 0 100 100">
+            <circle class="path" cx="50%" cy="50%" r="10" fill="none" stroke-width="2" stroke-miterlimit="10" />
+          </svg>
+        </div>
+        <ng-container *ngIf='!showSpinner'>
```

Comment:
instead of recreating the graph, it should show spinner on top of current graph and update dataset of current graph object